sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1770,7 +1770,7 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
    * Create a [[CalendarInterval]] for a unit value pair. Two unit configuration types are
    * supported:
    * - Single unit.
-   * - From-To unit (only 'YEAR TO MONTH' and 'DAY TO SECOND' are supported).
+   * - From-To unit (only 'YEAR TO MONTH' and 'DAY TO SECOND' and 'HOUR to SECOND' are supported).
```

Comment:
It sounds like we only added a very specific case. https://docs.teradata.com/reader/S0Fw2AVH8ff3MDA0wDOHlQ/RADkJCor1nDoyeD2T1VX5A 

Any reason?

Reply:
This is because `INTERVAL ... HOUR to SECOND` are most used in our existing scripts. If needed, maybe other cases like `INTERVAL .. X to X` can be contributed.